contracts/oracles/PythAggregatorV3.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "../utils/SolanaDataConverterLib.sol";
+import "../precompiles/QueryAccount.sol";
+
+
+/// @author https://twitter.com/mnedelchev_
+/// @custom:oz-upgrades-unsafe-allow constructor
+contract PythAggregatorV3 is OwnableUpgradeable, UUPSUpgradeable {
+    using SolanaDataConverterLib for bytes;
+    using SolanaDataConverterLib for uint32;
+    using SolanaDataConverterLib for uint64;
+
+    bytes32 public priceId;
+    bytes32 public priceFeedSolanaAccount;
+
+    error InvalidPriceFeedData();
+
+    /// @notice Disabling the initializers to prevent the implementation getting hijacked
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(bytes32 _priceId, bytes32 _priceFeedSolanaAccount) public initializer {       
+        __Ownable_init(msg.sender);
+
+        priceId = _priceId;
+        priceFeedSolanaAccount = _priceFeedSolanaAccount;
+    }
+
+    function _authorizeUpgrade(address) internal override onlyOwner {}
+
+    function _getPythDataFromSolana() internal view returns(int64, int64, int32) {
+        (bool success, bytes memory data) = QueryAccount.data(
+            uint256(priceFeedSolanaAccount), 
+            0, 
+            134
+        );
+        require(success, InvalidPriceFeedData());
+
+        int64 price = (data.toUint64(73)).readLittleEndianSigned64();
```

Comment:
@0xlucian you're right about this that it's a bit messier to read now and the change I will make is to add comments. These values cannot be constant or immutable, because the whole idea is to upgrade them if Pyth change the account structure in the future. If I make them as editable variables then in future upgrade I will also have to make updating methods and call them after the implementation upgrade, now if I leave them like this I will only have to upgrade the implementation.